### ISSUE TITLE
pdf-tools-install doesn't work on Mac OS

### ISSUE THREAD

**Issue by ashkan-leo** [politza/pdf-tools#645]
Hello

I get the following error when calling `pdf-tools-install` on mac os. I have brew and nix installed on my machine and was wondering maybe this is the reason pdf-tools break. I have all the dependencies installed so I'm totally lost here.

```
-*- mode: compilation; default-directory: "~/.emacs.d/.local/straight/build-27.1/pdf-tools/build/server/" -*-
Comint started at Thu Jan 21 20:48:02

/Users/ashkanaleali/.emacs.d/.local/straight/build-27.1/pdf-tools/build/server/autobuild -i /Users/ashkanaleali/.emacs.d/.local/straight/build-27.1/pdf-tools/
---------------------------
    Installing packages    
---------------------------
brew install pkg-config poppler automake
Error: 
  homebrew-cask is a shallow clone.
To `brew update`, first run:
  git -C /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask fetch --unshallow
This command may take a few minutes to run due to the large size of the repository.
This restriction has been made on GitHub's request because updating shallow
clones is an extremely expensive operation due to the tree layout and traffic of
Homebrew/homebrew-core and Homebrew/homebrew-cask. We don't do this for you
automatically to avoid repeatedly performing an expensive unshallow operation in
CI systems (which should instead be fixed to not use shallow clones). Sorry for
the inconvenience!
Warning: pkg-config 0.29.2_3 is already installed and up-to-date
To reinstall 0.29.2_3, run `brew reinstall pkg-config`
Warning: automake 1.16.3 is already installed and up-to-date
To reinstall 1.16.3, run `brew reinstall automake`
==> Downloading https://homebrew.bintray.com/bottles/libffi-3.3_2.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/839a830ef3a58dac841dbb07ed44b21993df707235819b44aeec3a4214001bfb--libffi-3.3_2.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/openssl%401.1-1.1.1i.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/96a5853f13074094ebd917dbbd52884a355d74adcd7c0adb3d67b1cb1a9d6574--openssl@1.1-1.1.1i.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/readline-8.1.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/b03c5b80e59c91f05f4327bf3cb7a4dbab63a8902b76c0c53c6d36eb2e4331e9--readline-8.1.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/sqlite-3.34.0.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/1a0242038ca8a6ce639d3a29ea0423dbd3a191255da3e4bb85add3e89608959a--sqlite-3.34.0.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/tcl-tk-8.6.11.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/b0dd2c101a5f383052e7e9e547907d507bc07c67b3e9ab7ad3a7ec7bbaf329dd--tcl-tk-8.6.11.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/python%403.9-3.9.1_6.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/34931aadcbed2d44dcad00df996913c7d7eab410defd1e6569225ba8c996db4f--python@3.9-3.9.1_6.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/glib-2.66.4_1.big_sur.bottle.1.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/74582d1533ae4484a7c1d648cd2eec90608a0ff800426dae7d81768e9fc6cfd1--glib-2.66.4_1.big_sur.bottle.1.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/cairo-1.16.0_4.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/e82ca05a8c5b6636d3db5b35a98bb17201cbcaa3a5e24a1a95899e57f96a7915--cairo-1.16.0_4.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/libtiff-4.2.0.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/83500ea467741e7ad6d0c48de97a930324fe222043e6126fb313a87e674c9092--libtiff-4.2.0.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/nss-3.60.1.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/6de5393f031d95ac922e8aa00d111d38bf1baa7a61b85c08aa267f6702619b23--nss-3.60.1.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/qt-5.15.2.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/114bb5dda976c58aa2bc7c8a9aefa87b6adcc25f668d5c5ccdd2c9aec310dfad--qt-5.15.2.big_sur.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/bottles/poppler-21.01.0.big_sur.bottle.tar.gz
Already downloaded: /Users/ashkanaleali/Library/Caches/Homebrew/downloads/54409d6a8911b9e209c5cb7017523fbb364e20f28b8b753ef223f558e18d452f--poppler-21.01.0.big_sur.bottle.tar.gz
Error: poppler 20.11.0 is already installed
To upgrade to 21.01.0, run `brew upgrade poppler`.

---------------------------
 Configuring and compiling 
---------------------------
./configure -q --bindir=/Users/ashkanaleali/.emacs.d/.local/straight/build-27.1/pdf-tools/ && make -s
./configure: line 5514: syntax error near unexpected token `png,'
./configure: line 5514: `PKG_CHECK_MODULES(png, libpng)'
===========================
     Build failed.  ;o(    
===========================
Note: maybe try the '-d' option.

Comint exited abnormally with code 1 at Thu Jan 21 20:48:15

```

**Comment by ashkan-leo** [comment]
`autobuild --os nixos ` also gives the same error

**Comment by ashkan-leo** [comment]
I'm still unable to figure this out...

**Comment by kontrapant** [comment]
im getting a very similar error. I wasn't able to build pdf-tools since updateing to macOS 11, but im not sure if this is the cause or if it might have something to do with the current `doom emacs` version im using

```
-*- mode: compilation; default-directory: "~/.emacs.d/.local/straight/build-27.1/pdf-tools/build/server/" -*-
Comint started at Mon Feb  1 14:18:37

/Users/Vinc/.emacs.d/.local/straight/build-27.1/pdf-tools/build/server/autobuild -i /Users/Vinc/.emacs.d/.local/straight/build-27.1/pdf-tools/
---------------------------
    Installing packages    
---------------------------
brew install pkg-config poppler automake

Warning: pkg-config 0.29.2_3 is already installed and up-to-date.
To reinstall 0.29.2_3, run:
  brew reinstall pkg-config
Warning: poppler 21.01.0 is already installed and up-to-date.
To reinstall 21.01.0, run:
  brew reinstall poppler
Warning: automake 1.16.3 is already installed and up-to-date.
To reinstall 1.16.3, run:
  brew reinstall automake

---------------------------
 Configuring and compiling 
---------------------------
./configure -q --bindir=/Users/Vinc/.emacs.d/.local/straight/build-27.1/pdf-tools/ && make -s
configure: error: Package requirements (libpng) were not met:

No package 'libpng' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables png_CFLAGS
and png_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
===========================
     Build failed.  ;o(    
===========================
Note: maybe try the '-d' option.

Comint exited abnormally with code 1 at Mon Feb  1 14:18:53
```

\
\
I located all `libpng.pc` files and added the paths to `PKG_CONFIG_PATH`,  which now looks like this:

```
$ echo $PKG_CONFIG_PATH
/usr/local/lib/pkgconfig:/usr/local/Cellar/libpng/1.6.37/lib/pkgconfig:/opt/X11/lib/pkgconfig:/System/Volumes/Update/mnt1/usr/local/lib/pkgconfig:/System/Volumes/Update/mnt1/opt/X11/lib/pkgconfig
```
this doesnt seem to have any effect though
\
I don't understand how or where to set the `png_CFLAGS` or `png_LIBS` variables, i only found info about them as flags in the manpage for pkg-config

\
\
a doom emacs update as well as an macOS 11.2 update just came in,
will report back after installation

**Comment by kontrapant** [comment]
os and doom update sadly didn't solve the problem :/

**Comment by JudicialDomicide** [comment]
I'm also experiencing this same issue on macbook pro running Mojave 10.14.6 and Emacs version 27.1 installed via brew cask

On opening emacs I get a Need to (re)build the epdinfo program do it now. I choose (y) and it fails.

`-*- mode: compilation; default-directory: "~/.emacs.d/elpa/pdf-tools-20200512.1524/build/server/" -*-
Comint started at Mon Feb  1 17:29:50

/Users/$user/.emacs.d/elpa/pdf-tools-20200512.1524/build/server/autobuild -i /Users/$user/.emacs.d/elpa/pdf-tools-20200512.1524/
Failed to recognize this system, trying to continue.
---------------------------
 Configuring and compiling 
---------------------------
./configure -q --bindir=/Users/$user/.emacs.d/elpa/pdf-tools-20200512.1524/ && make -s
configure: error: in `/Users/$user/.emacs.d/elpa/pdf-tools-20200512.1524/build/server':
configure: error: The pkg-config script could not be found or is too old.  Make sure it
is in your PATH or set the PKG_CONFIG environment variable to the full
path to pkg-config.

Alternatively, you may set the environment variables png_CFLAGS
and png_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

To get pkg-config, see <http://pkg-config.freedesktop.org/>.
See `config.log' for more details
===========================
     Build failed.  ;o(    
===========================
Note: maybe try the '-d' option.

Comint exited abnormally with code 1 at Mon Feb  1 17:29:52`

**Comment by JudicialDomicide** [comment]
UPDATE: It is definitely related to Enviroment Path issues I think with /.zshrc file

This is a work-around until someone smarter than me can figure out what is going on.

open Terminal
```
 /Applications/Emacs.app/Contents/MacOS/Emacs  
```

**Comment by kontrapant** [comment]
This workaround doesn't work for me.
I tried moving the **PKG_CONFIG_PATH** from `.zprofile` to `.zshrc` to `.zshenv`.
I also installed the `exec-path-from-shell` package to be on the same baselevel.
Sadly im getting the same error as before; when launching emacs from terminal or when `./autobuild`ing directly in the terminal

**Comment by JudicialDomicide** [comment]
@kontrapant If you cant ./autobuild **directly in the terminal** (irrespective of emacs) then it would suggest that libpng is not installed correctly??

Perhaps try the following steps:
(assuming using brew) remove the libpng framework in /Library/Frameworks and do a
`brew install libpng --universal`

and force the link

`brew link libpng --force`

then retry in terminal and see what happens. I wish you luck.

**Comment by kontrapant** [comment]
reinstall didn't solve the issue for me, but you're right, apparently the problem lies not in emacs but in macOS generally not finding `libpng` 

thanks for pointing me in the right direction :)

**Comment by ashkan-leo** [comment]
Here is a workaround that worked for me. I use nixpkgs in Mac OS.

```bash
$ cd ~/.emacs.d/.local/straight/build-27.1/pdf-tools/build/server
$ nix-shell -p pkg-config poppler automake libtool libpng
$ autoreconf -i -f
$ ./autobuild -i $HOME/.emacs.d/.local/straight/build-27.1/pdf-tools/ --os nixos
```

I think the key parts that I was missing was the `libtool` dependency and the `autoreconf -i -f` command.  

Specifically, the following error is caused by the lack of `libtool` which includes the `PKG_CHECK_MODULES` macro expansion.
```
./configure: line 5514: syntax error near unexpected token `png,'
./configure: line 5514: `PKG_CHECK_MODULES(png, libpng)'
```

**Comment by wm-oci** [comment]
> Here is a workaround that worked for me. I use nixpkgs in Mac OS.
> 
> ```shell
> $ cd ~/.emacs.d/.local/straight/build-27.1/pdf-tools/build/server
> $ nix-shell -p pkg-config poppler automake libtool libpng
> $ autoreconf -i -f
> $ ./autobuild -i $HOME/.emacs.d/.local/straight/build-27.1/pdf-tools/ --os nixos
> ```


I had to add `autoconf` to the list of `-p` arguments to `nix-shell` in addition to `automake`, and then `autoreconf` was available within `nix-shell`, and it then it worked as advertised.

**Comment by gveres** [comment]
I've ran into the same issue on MacOS 11.2.1. I've found that the root cause is how homebrew installs parts of the build chain used by pdf-tools. Essentially, the make tools could not resolve the PKG_CHECK_MODULES macro, because its definition is not where it's expected to be. My solution was to add the directory in which this macro is defined (in the file pkg.m4) to the list of directories that are used to resolve macros during the process.
In practice:
`sudo echo "/usr/local/Cellar/pkg-config/0.29.2_3/share/aclocal" >> /usr/local/Cellar/automake/1.16.3/share/aclocal/dirlist`
was the command that fixed the build process for me. Note, your versions might differ. The main thing is that your brew installed automake needs to know about your brew installed pkg-config's macro defs.
After this, the build process went smoothly from within Emacs.
I hope this helps.
